module/testing.py

```diff
@@ -27,10 +28,31 @@ def lock_path(display):
     return '/tmp/.X%d-lock' % display
 
 
+def get_lock(display):
+    lockfile = lock_path(display)
+    try:
+        with open(lockfile, "w") as f:
+            fcntl.flock(f.fileno(), fcntl.LOCK_EX | fcntl.LOCK_NB)
+        return True  # successfully got a lock
+    except OSError:
+        return False  # failed to get a lock
+
+
+def release_lock(display):
+    lockfile = lock_path(display)
+    if os.path.exists(lockfile):
+        try:
+            with open(lockfile, "r") as f:
+                fcntl.flock(f.fileno(), fcntl.LOCK_UN)
+        except OSError:
+            pass
+        os.remove(lockfile)
```

Comment:
Not sure about that. Should I pass the exception? Or let it bubble up?

Reply:
I think this seems fine. One could argue that we should move it inside the above except:, since if we can't unlock the file we probably shouldn't remove it. But it seems unlikely either way.